doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+portable across multiple array libraries and their supported features like GPUs
+and other hardware accelerators or JIT compilers.
+
+``device`` support
+^^^^^^^^^^^^^^^^^^
+
+Device support is perhaps the most obvious example. NumPy is and will remain a
+CPU-only library, so why bother introducing a ``ndarray.device`` attribute or
+``device=`` keywords in several functions? This one feature is purely meant to
+make it easier to write code that is portable across libraries. The ``.device``
+attribute will return an object representing CPU, and that object will be
+accepted as an input to ``device=`` keywords. For example:
+
+.. code::
+
+    # Should work when `xp` is `np` and `x1` a numpy array
+    x2 = xp.asarray([0, 1, 2, 3], dtype=xp.float64, device=x1.device)
+
+This will work as expected for NumPy, creating a 1-D numpy array from the input
+list. It will also work for CuPy & co, where it may create a new array on a GPU
+or other supported device.
+
+
+``isdtype``
+^^^^^^^^^^^
+
+The array API standard introduced a new function ``isdtype`` for introspection
+of dtypes, because there was no suitable alternative in NumPy. The closest one
+is ``np.issubdtype``, however that assumes a complex class hierarchy which
+other array libraries don't have, isn't the most ergonomic API, and required a
+larger API surface (``np.floating`` and friends). ``isdtype`` will be the new
+and canonical way to introspect dtypes. All it requires from a dtype is that
+``_eq__`` is implemented and has the expected behavior when compared with other
+dtypes from the same library.
+
+Note that as part of the effort on NEP 52, some dtype aliases were removed and
+canonical Python and C names documented. See also `gh-17325
+<https://github.com/numpy/numpy/issues/17325>`__ covering issues with NumPy's
+lack of a good API for this.
+
+
+``copy`` keyword semantics
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``copy`` keyword in ``asarray`` and ``array`` will now support
+``True``/``False``/``None`` with new meanings:
+
+- ``True`` - Always make a copy.
+- ``False`` - Never make a copy. If a copy is required a ``ValueError`` is raised.
+- ``None`` - A copy will only be made if it is necessary (previously ``False``).
+
+The ``copy`` keyword in ``astype`` will stick to its current meaning, because
+"never copy" when asking for a cast to a different dtype doesn't quite make
+sense.
+
+
+New function name aliases
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the Python API cleanup for NumPy 2.0 (see :ref:`NEP52`) we spent a lot of
+effort removing aliases. So introducing new aliases has to have a good
+rationale. In this case, it is needed in order to match other libraries.
+The main set of aliases added is for trigonometric functions, where
+the array API standard chose to follow C99 and other libraries in using
+``acos``, ``asin`` etc. rather than ``arccos``, ``arcsin``, etc. NumPy usually
+also follows C99; it is not entirely clear why this naming choice was made many
+years ago.
```

Comment:
minor
```suggestion
``acos``, ``asin`` etc. rather than ``arccos``, ``arcsin``, etc. NumPy also
usually follows C99; it is not entirely clear why this naming choice was made many
years ago.
```